server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -759,7 +761,7 @@ public IPAddressVO doInTransaction(TransactionStatus status) throws Insufficient
 
                 Filter filter = new Filter(IPAddressVO.class, "vlanId", true, 0l, 1l);
 
-                List<IPAddressVO> addrs = _ipAddressDao.lockRows(sc, filter, true);
+                List<IPAddressVO> addrs = _ipAddressDao.search(sc, filter, false);
```

Comment:
The issue seems to be here where search happens without no locks and other threads may end up getting the same list of non-free ip addresses to allocate address from.